test/unit/application.spec.js

```diff
@@ -65,20 +65,6 @@ describe('marionette application', function() {
     });
   });
 
-  describe('when an app has been started, and registering another initializer', function() {
```

Comment:
Code can always be refactored to work without Callbacks. Just use Promises, as in @megawac's example. If there's any specific code you'd like us to look at @derickbailey, you should post a Gist!